l10n_eu_product_adr/models/adr_label.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 Iryna Vyshnevska (Camptocamp)
+# Copyright 2021 Stefan Rijnhart <stefan@opener.amsterdam>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AdrLabel(models.Model):
+    _name = "adr.label"
+    _description = "Dangerous Goods Label"
+    _order = "code, name"
+
+    name = fields.Char(required=True, translate=True)
+    code = fields.Char(required=True)
+    image = fields.Binary(required=True)
+    class_id = fields.Many2one(comodel_name="adr.class", string="Class", required=True)
+
+    def unlink(self):
+        """Restrict removal of labels in use"""
+        for label in self:
+            if label.class_ids:
```

Comment:
```suggestion
            if label.class_ids:
```
I see class_id not class_ids and `ondelete='restrict'` should solve this issue

Reply:
Thanks, I missed this at some point during the refactoring. I've changed this to check the linked goods, and added tests for this method and others.